src/box/msgpack.c

```diff
@@ -143,6 +144,14 @@ msgpack_check_ext_data(int8_t type, const char *data, uint32_t len)
 			return -1;
 		}
 		return 0;
+	case MP_ARROW:
+		if (mp_validate_arrow(data, len) != 0) {
+			/* The error is set by arrow_ipc_decode(). */
+			diag_add(ClientError, ER_INVALID_MSGPACK,
+				 "cannot unpack arrow data");
```

Comment:
Looks like this new check broke `go-tarantool` integration tests:

```
=== RUN   TestInsert_invalid
=== RUN   TestInsert_invalid/0
    tarantool_test.go:76: 
        	Error Trace:	/home/runner/work/tarantool/tarantool/arrow/tarantool_test.go:76
        	Error:      	Error "Invalid MsgPack - packet body (ClientError, code 0x14), see ./src/box/xrow.c line 278: Invalid MsgPack - invalid extension (ClientError, code 0x14), see ./src/box/msgpack.c line 178: Invalid MsgPack - cannot unpack arrow data (ClientError, code 0x14), see ./src/box/msgpack.c line 150: Arrow decode error: unexpected data size (DecodeError, code 0x20), see ./src/lib/core/arrow_ipc.c line 95" does not contain "Failed to decode Arrow IPC data"
        	Test:       	TestInsert_invalid/0
=== RUN   TestInsert_invalid/1
    tarantool_test.go:76: 
        	Error Trace:	/home/runner/work/tarantool/tarantool/arrow/tarantool_test.go:76
        	Error:      	Error "Invalid MsgPack - packet body (ClientError, code 0x14), see ./src/box/xrow.c line 278: Invalid MsgPack - invalid extension (ClientError, code 0x14), see ./src/box/msgpack.c line 178: Invalid MsgPack - cannot unpack arrow data (ClientError, code 0x14), see ./src/box/msgpack.c line 150: Arrow decode error: Expected at least 8 bytes in remainder of stream (DecodeError, code 0x20), see ./src/lib/core/arrow_ipc.c line 137" does not contain "Failed to decode Arrow IPC data"
        	Test:       	TestInsert_invalid/1
=== RUN   TestInsert_invalid/2
--- FAIL: TestInsert_invalid (0.00s)
    --- FAIL: TestInsert_invalid/0 (0.00s)
    --- FAIL: TestInsert_invalid/1 (0.00s)
    --- PASS: TestInsert_invalid/2 (0.00s)
FAIL
FAIL	github.com/tarantool/go-tarantool/v2/arrow	0.116s
```

[CI link](https://github.com/tarantool/tarantool/actions/runs/11476215216/job/31936059576?pr=10665)

Reply:
Waiting for:
- https://github.com/tarantool/go-tarantool/issues/415